_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+By merging the responsibilities of a `UIView` and `UIViewController` into a Component, there is a consistent way to work with all aspects of your app. 
+
+To try to undestand this, let's take a trivial example. Downloading some data from the network and showing it on a screen.
+
+In UIKit-world you would:
+
+-   Create a `UIViewController` which makes the API request on its `viewDidLoad`
+-   While the request is sent you present a set of views for loading
+-   When the API request has returned you remove the loading screen
+-   You take the data from the request and create a view hierarchy then present that
+
+In React you would:
+
+-   Create a component which makes the API request on its `onMount`
+-   While the request is sent you render another component for showing loading
+-   The results come back and you change your "state" on the main component with the API request
+-   The state change re-runs your render method, which passes the API "state" down to the component for your page
+
+They are conceptually very similar. React does two key things differently: Handle "state" changes on any component, and handle view creation/addition and removal.
+
+## Handling "State" Changes
+
+So, I've been quoting "state", I should explain this. There are two types of "state" inside React, and I've been using the quoted term to refer to both for simplicity till now.
+
+> There are two types of data that control a component: `props` and `state`. `props` are set by the parent and they are fixed throughout the lifetime of a component. For data that is going to change, we have to use `state`.
+
+> [React Native - State](https://facebook.github.io/react-native/docs/state.html)
+
+So in our case above, getting the API results only changes the state on the component which makes the request. However, the results are passed down into the props _(properties)_ of the component's children as any further changes to the API data (for example if you were polling for updates) would result in a re-render of the child-components.
+
+So for the lifetime of that top-level component, the changes due to the API request are put in state. Then the results are passed down to it's children as props. This means the children can potentially change when an API response is received.
+
+## Handling View Management
+
+Because of the consolidated rules around state management React can quite easily know when there have been changes throughout your component tree and to call `render` for those components. `render` is the function where you declare the tree of children for a component.
+
+> The flow in React is one-directional. We maintain a hierarchy of components, in which each component depends only on its parent and its own internal state. We do this with properties: data is passed from a parent to its children in a top-down manner. If an ancestor component relies on the state of its descendant, one should pass down a callback to be used by the descendant to update the ancestor.
```

Comment:
"The flow" -> what flow? You can modify the quote and surround your changes with [square brackets] for explicitness.